src/eval_framework/llm/models.py

```diff
@@ -240,22 +270,173 @@ def __init__(self, model_name: Optional[str] = None, formatter: str = "Llama3For
         self.model = AutoModelForCausalLM.from_pretrained(self.LLM_NAME, device_map="auto")
 
         # Lazy formatter initialization - only create the one we need
-        selected_formatter = self._get_formatter(formatter, model_name)
+        selected_formatter = FormatterFactory.create_formatter(formatter, model_name)
 
         print(f"{RED}[ Model initialized --------------------- {RESET}{self.LLM_NAME} {RED}]{RESET}")
         print(f"{RED}[ Formatter: {formatter} ]{RESET}")
         self._set_formatter(selected_formatter)
 
-    def _get_formatter(self, formatter: str, model_name: str) -> Any:
-        """Get formatter instance based on formatter name."""
-        if formatter == "Llama3Formatter":
-            return Llama3Formatter()
-        elif formatter == "MistralFormatter":
-            return MagistralFormatter(model_name)
-        elif formatter == "ConcatFormatter":
-            return ConcatFormatter()
-        elif formatter == "HFFormatter":
-            return HFFormatter(model_name)
+
+class HFLLM_from_wandb_registry(HFLLM):
+    """
+    A class to create HFLLM instances from registered models in Wandb registry.
+    Downloads the model artifacts from Wandb and creates a local HFLLM instance.
+    """
+
+    def __init__(
+        self,
+        artifact_name: str,
+        version: str = "latest",
+        formatter: str = "",
+        formatter_identifier: str = "",
+        **kwargs: Any,
+    ) -> None:
+        """
+        Initialize HFLLM from a Wandb registered model artifact.
+
+        Args:
+            artifact_name: Name of the artifact in the Wandb registry
+            version: Version of the artifact to download (default: "latest")
+            formatter: Type of formatter to use (default: "")
+            **kwargs: Additional arguments passed to the parent class
+        """
+        print(f"{RED}[ Loading registered model from Wandb: {artifact_name}:{version} ]{RESET}")
+
+        with WandbFs() as wandb_fs:
+            artifact = wandb_fs.get_artifact(artifact_name, version)
+            file_list = wandb_fs.ls(artifact)
+            wandb_fs.download_artifacts(artifact)
+            file_root = wandb_fs.find_hf_checkpoint_root_from_path_list(file_list)
```

Comment:
1) What is the motivation for this? What are the possible supported structures of checkpoints that one needs to search for the root?

2) The function `find_hf_checkpoint_root_from_path_list()` and the other function it calls internally is rather hard to comprehend. First of all, the artifact is downloaded in its entirety and only then the root is being searched for. So IMHO `find_hf_checkpoint_root_from_path_list()` can be much simpler as it can operate on local file system (and not on s3)? Could it be as simple as `wandb_fs.temp_dir.glob("**/config.json")`?